### PR TITLE
Enable ucx in legate.core builds.

### PR DESCRIPTION
**Pull request by mag1cp1n**
- Does not enable ucx for MAC.

### CODE REVIEW

File: conda/conda-build/meta.yaml [hunk 1]
```diff
@@ -53,6 +61,12 @@
 {% set debug_tag='' %}
 {% endif %}
 
+{% if ucx_configured_bool %}
+{% set ucx_tag='_ucx' %}
```
Comment (m3vaz) [inline conda/conda-build/meta.yaml:65]: indentation?

File: continuous_integration/scripts/build-legate-conda [hunk 2]
```diff
@@ -58,8 +61,15 @@ cmake_version:
...
 package_version:
   - "$(git -C $REPO_DIR describe --abbrev=0 --tags | $SED 's/[a-zA-Z]//g' | cut -d '.' -f -2).00"
+ucx_configured:
+  - "${UCX_CONFIGURED}"
 EOF
-
+if [ "$UCX_ENABLED" = "ON" ]; then
+    cat <<EOF >> $REPO_DIR/conda/conda-build/conda_build_config.yaml
+ucx:
+  - ">=1.14"
```
Comment (m3vaz) [inline continuous_integration/scripts/build-legate-conda:70]: >=1.14? is there a reason we pick 1.14 specifically?

File: continuous_integration/scripts/build-legate-cpp [hunk 3]
```diff
@@ -11,7 +11,7 @@ build_legate_cpp() {
 
     cmake_args+=(-DCMAKE_BUILD_TYPE=Release);
     cmake_args+=(-DLegion_CUDA_ARCH=all-major);
-    cmake_args+=(-DLegion_NETWORKS=);
+    cmake_args+=(-DLegion_NETWORKS=${UCX_ENABLED:-ucx})
```
Comment (m3vaz) [inline continuous_integration/scripts/build-legate-cpp:14]: semicolon?

**Review by m3vaz — CHANGES_REQUESTED** [review]

File: conda/conda-build/meta.yaml (hunk 1 above)
Comment (mag1cp1n) [reply conda/conda-build/meta.yaml:65]: Sorry, Didn't get this. It follows the other settings.

File: continuous_integration/scripts/build-legate-conda (hunk 2 above)
Comment (mag1cp1n) [reply continuous_integration/scripts/build-legate-conda:70]: As per mentioned here: https://github.com/nv-legate/legate.core/blob/branch-24.01/BUILD.md

File: continuous_integration/scripts/build-legate-cpp (hunk 3 above)
Comment (mag1cp1n) [reply continuous_integration/scripts/build-legate-cpp:14]: Done.

Comment (mag1cp1n) [reply continuous_integration/scripts/build-legate-cpp]: Done.

File: continuous_integration/scripts/build-legate-conda (hunk 2 above)
Comment (m3vaz) [reply continuous_integration/scripts/build-legate-conda:70]: ah, makes sense, then that number should probably also be reflected in the meta.yaml as well? (I'll make a separate comment there)

File: conda/conda-build/meta.yaml [hunk 4]
```diff
@@ -144,6 +158,9 @@ requirements:
     - libcurand-dev
     - libcusolver-dev
 {% endif %}
+{% if ucx_configured_bool %}
+    - ucx
```
Comment (m3vaz) [inline conda/conda-build/meta.yaml:162]: should this be >=1.14 (per BUILD.md)

File: conda/conda-build/meta.yaml (hunk 1 above)
Comment (m3vaz) [reply conda/conda-build/meta.yaml:65]: ah it does :(

**Review by m3vaz — APPROVED** [review]

**Pull request merged**